File: Tools/Postprocessing/F_Src/fvarnames.f90
! simply print out the list of variables stored in a plotfile
program fvarnames

  use plotfile_module
  use filler_module
  use bl_IO_module

  implicit none

  integer n, narg
  integer :: unit
  character (len=256) :: fname

  type(plotfile) :: pf

100 format (1x, i3, 3x, a)

  ! parse any formatting arguments
  narg = command_argument_count()

  if ( narg .eq. 0 ) then
     print *, ''
     print *, 'Usage:'
     print *, '     ftime plotfile'
     print *, ''
     print *, 'Description:'
     print *, '     This program takes a single plotfile and dumps out the list of variables'
     print *, ''
     stop
  endif

  unit = unit_new()

  call get_command_argument(1, value = fname)
  call build(pf, fname, unit)

  do n = 1, pf%nvars
     write(*,100) n, trim(pf%names(n))
  enddo

  call destroy(pf)

end program fvarnames

File: Tools/Plotfile/fvarnames.cpp
#include <AMReX.H>
#include <AMReX_Print.H>
#include <AMReX_PlotFileUtil.H>

using namespace amrex;

void main_main()
{
    const int narg = amrex::command_argument_count();

    if (narg == 0) {
        amrex::Print()
            << "\n"
            << " Usage:\n"
            << "      fvarnames plotfile\n"
            << "\n"
            << " Description:\n"
            << "      This program takes a single plotfile and dumps out the list of variables\n"
            << std::endl;
        return;
    }

    const auto& fname = amrex::get_command_argument(1);
    PlotFileData plotfile(fname);
    const auto& names = plotfile.varNames();
    int n = 0;
    for (auto const& name : names) {
        amrex::Print() << std::setw(4) << n++ << "   " << name << "\n";
    }
}

int main (int argc, char* argv[])
{
    amrex::SetVerbose(0);
    amrex::Initialize(argc, argv, false);
    main_main();
    amrex::Finalize();
}
